### DISCUSSION TITLE
Add Editor Settings to Configure Default Editor Naming Values for New Projects

### DISCUSSION THREAD

**Discussion by Anixias** [godotengine/godot-proposals#9167] [Editor]
Hi, all.

I make new projects in Godot very frequently. Every single time, I have to immediately go to the Project Settings and modify most properties under `editor/naming/` because I use different naming conventions than the default ones. This could be easily improved by adding the same settings to the Editor Settings to be the default values when creating a new project, assuming that the project creation process has access to the editor settings.

Alternatively, it could be nice instead to have a new project configuration step where you can select a config file that defines new default values for any project settings you wish, which would be more powerful than the above suggestion but obviously more work to implement. The config file could be a simple format that is simply a list of key-value pairs (or JSON, if that is somehow easier), like this:

```
editor/naming/default_signal_callback_name: "On{NodeName}{SignalName}"
editor/naming/default_signal_callback_to_self_name: "On{SignalName}"
editor/naming/scene_name_casing: "PascalCase"
```

If I knew anything about how to modify the engine (it's too overwhelming to me, honestly), I'd implement this myself. If anyone needs help to lex and parse the custom config format, I am able to assist with that. Would love to know what others think of this idea.

**Comment by Calinou** [comment]
I feel a general-purpose project templates system would be more suited for this: https://github.com/godotengine/godot-proposals/issues/3029

This is also because you likely want to use different conventions in GDScript and C# projects.
